src/model/markerscollection.js

```diff
@@ -0,0 +1,142 @@
+ *
+ * Markers are simply {@link module:engine/model/liverange~LiveRange live ranges} added to `MarkersCollection`.
+ * Markers are used to represent information connected with model document. In contrary to
+ * {@link module:engine/model/node~Node nodes}, which are bits of data, markers are marking a part of model document.
+ * Each live range/marker is added with `name` parameter. Name is used to group and identify markers. Multiple live
+ * ranges can be added under the same name.
+ *
+ * Whenever live range is added or removed from markers collection, markers collection fires `addMarker` and `removeMarker`
+ * events. Same happens when a live range is changed due to changes in the model (both events are fired, `removeMarker` first,
+ * then `addMarker`).
+ *
+ * Markers can be converted to view by adding appropriate converters for
+ * {@link module:engine/conversion/modelconversiondispatcher~ModelConversionDispatcher#event:addMarker addMarker} and
+ * {@link module:engine/conversion/modelconversiondispatcher~ModelConversionDispatcher#event:removeMarker removeMarker}
+ * events, or by building converters for {@link module:engine/conversion/modelconversiondispatcher~ModelConversionDispatcher}
+ * using {@link module:engine/conversion/buildmodelconverter~buildModelConverter model converter builder}.
+ *
+ * Markers are similar to adding and converting attributes on nodes. The difference is that attribute is connected to
+ * a given node (i.e. a character is bold no matter if it gets moved or content around it changes). Markers on the
+ * other hand are continuous ranges (i.e. if a character from inside of marker range is moved somewhere else, marker
+ * range is shrunk and the character does not have any attribute or information that it was in the marked range).
+ *
+ * @memberOf engine.model
+ */
+export default class MarkersCollection {
+	/**
+	 * Creates a markers collection.
+	 */
+	constructor() {
+		/**
+		 * Stores range to marker name bindings for added ranges.
+		 *
+		 * @private
+		 * @member {Map} #_markerNames
+		 */
+		this._markerNames = new Map();
+	}
+
+	/**
+	 * Sets a name for given live range and adds it to the markers collection. Multiple live ranges may have same name.
+	 * Markers can also be grouped while still having different names, i.e.: `search:22`, `search:37`.
+	 *
+	 * Throws, if given `liveRange` is not an instance of {@link module:engine/model/liverange~LiveRange}.
+	 *
+	 * Throws, if given {@link module:engine/model/liverange~LiveRange LiveRange} instance is already added to the collection.
+	 *
+	 * `MarkersCollection` instance listens to `change` event of all live ranges added to it and fires `removeMarker`
+	 * and `addMarker` events when the live range changes.
+	 *
+	 * @param {module:engine/model/liverange~LiveRange} liveRange Live range to be added as a marker to markers collection.
+	 * @param {String} markerName Name to be associated with given `liveRange`.
+	 */
+	addRange( liveRange, markerName ) {
```

Comment:
It's markers collection why I'm adding `range`? I'm adding range with a name so marker (even `addMarker` event is fired). So `addMarker` would be better. But since there is nothing else I can add maybe simple `add`?

Also, it's more natural to have a name as the first parameter, so:

`add( name, range )`?